datafusion/core/tests/sql/select.rs

```diff
@@ -1001,3 +1001,41 @@ async fn query_empty_table() {
     let expected = vec!["++", "++"];
     assert_batches_sorted_eq!(expected, &result);
 }
+
+#[tokio::test]
+async fn case_insensitive_in_sql() -> Result<()> {
+    // Test that field name and table name in sql is case-insensitive
+    let fields = vec![
+        Field::new("Column1", DataType::Utf8, true),
```

Comment:
This table has schema equivalent to doing the following (note the `"` are very important as they say to not normalize case):

```sql
CREATE TABLE test(
  "Column1" varchar,
  "COLUMN2" varchar,
  "column3" varchar
)
```

And in that case, I would expect 

```sql
"SELECT COLumn1, colUMN2, column3 FROM test WHERE TEst.COLUMN1='content1' and column2='content2' and COLumn3='content3'"
``` 

to fail. And in fact it does in postgres:

```sql
alamb=# CREATE TABLE test(
  "Column1" varchar,
  "COLUMN2" varchar,
  "column3" varchar
);
CREATE TABLE
alamb=# SELECT COLumn1, colUMN2, column3 FROM test WHERE TEst.COLUMN1='content1' and column2='content2' and COLumn3='content3';
ERROR:  column "column1" does not exist
LINE 1: SELECT COLumn1, colUMN2, column3 FROM test WHERE TEst.COLUMN...
               ^
HINT:  Perhaps you meant to reference the column "test.Column1" or the column "test.column3".
alamb=# 

```

Reply:
It seems like we should have had a unit test to catch the regression. I filed https://github.com/apache/arrow-datafusion/issues/2227 to add one.

Reply:
This PR is to fix the issue #2211 , so i make it case-insensitive to solve it. I think that if we want to keep it case-sensitive, we should solve the problem in another way.

Accroding to the error: No field named 'test.column1'. i think the field name was converted to lower-case and compare to the real filed name. If the real field name contains uppercase letters, it would always fail.

Reply:
Sorry -- I should have been more specific -- I don't think #2211 is a bug. I left a comment there explaining why: https://github.com/apache/arrow-datafusion/issues/2210#issuecomment-1099113763

Reply:
Thanks, i got it.